lib/draw.c

```diff
@@ -14,21 +13,299 @@
 #define DEBUG2 '2'
 #define DEBUG3 '3'
 
-void initPoints(double arr[8][3], double basePoints[8][3])
-{
-    for (int i = 0; i < 8; i++)
-    {
-        for (int j = 0; j < 3; j++)
-        {
+#define MAX_LINE_LENGTH 256
+
+mesh importMeshFromOBJFile (char * pathToFile){
+
+    FILE *obj = fopen(pathToFile, "r");
+
+    mesh newMesh;
+    mesh zerodMesh;
+    newMesh.numOfTris = 0;
+
+    if (NULL == obj){
+
+        printf("Error: .OBJ file not found\n");
+        printf("Try to run the program from top level \"Cube/\" dir\n");
+        return newMesh;
+    }
+
+    char line[MAX_LINE_LENGTH];
+    int verts = 0;
+    int faces = 0;
+    vector pointCoords;
+    vector average;
+    vector runningTotal;
+    triangle tempTriangle;
+    int p0, p1, p2;
+    
+    //counts faces and verticies.
+    while (fgets(line, sizeof(line), obj) != NULL){
+
+        if (line[0] == 'v' && line[1] == ' '){
+
+            verts++;
+        }else if (line[0] == 'f' && line[1] == ' '){
+
+            faces++;
+        }
+    }
+
+    newMesh.numOfTris = faces;
+    newMesh.numOfVerts = verts;
+
+    vector (*vectorArray) = malloc(newMesh.numOfVerts * sizeof(vector)); //dynamically allocates an array, based off the number points = (verticies * 3).
+
+    newMesh.tris = (triangle*) malloc(newMesh.numOfTris * sizeof(triangle)); //dynamically allocates memory for the number of triangles
+    zerodMesh = newMesh;
+
+    rewind(obj);
+
+    int vCount = 0;
+    int fCount = 0;
+    while (fgets(line, sizeof(line), obj) != NULL) {
+
+        //reads all lines of obj that start with "v " into the dynamically assigned array of points.
+        if (line[0] == 'v' && line[1] == ' '){
+
+            sscanf(line,"v %lf %lf %lf", &pointCoords.x, &pointCoords.y, &pointCoords.z);
+
+            vectorArray[vCount] = pointCoords;
+
+            vCount++;
+
+            runningTotal = addVec(runningTotal, pointCoords); //only represents summed total of each co-ord at this point.
+        }
+
+        //reads all lines of obj that start with "f ".
+        //each int after 'f ' represents an index (starting at 1), of the array of verticies.
+        //
+        else if (line[0] == 'f' && line[1] == ' '){
+
+            sscanf(line,"f %d %d %d", &p0, &p1, &p2);
+
+            newMesh.tris[fCount].p[0] = vectorArray[(p0-1)];
+            newMesh.tris[fCount].p[1] = vectorArray[(p1-1)];
+            newMesh.tris[fCount].p[2] = vectorArray[(p2-1)];
+
+            fCount++;
+        }
+    }
+
+    //Handy debug for seeing if points were imported properly
+    // for(int i = 0; i < vCount; i++){
+    //      printf("%d: %lf, %lf, %lf\n", i, vectorArray[i].x, vectorArray[i].y, vectorArray[i].z);
+    // }
```

Comment:
if adding debug code just wrap in a ifdef macro like this and have a single line at top of file with the macro commented out. 
i.e.
```suggestion

// #DEFINE DEBUG_POINTS

#ifdef DEBUG_POINTS
    //Handy debug for seeing if points were imported properly
     for(int i = 0; i < vCount; i++){
          printf("%d: %lf, %lf, %lf\n", i, vectorArray[i].x, vectorArray[i].y, vectorArray[i].z);
     }
#ENDIF
```